lesson26/src/js/login.js

```diff
@@ -0,0 +1,134 @@
+const userName = document.getElementById("username");
+const password = document.getElementById("password");
+const loginButton = document.getElementById("js-login-button");
+const storage = localStorage;
+
+if (storage.getItem("token")) window.location.href = "./index.html";
+
+
+const constraint = {
+  username: {
+    validation: () => {
+      return isLimitTextLength(userName.value, 16);
+    },
+    invalidMessage: "ユーザー名は15文字以下にしてください。"
+  },
+  password: {
+    validation: () => {
+      const reg = /^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])[a-zA-Z0-9]{8,}$/g;
+      return isValidInRegex(reg, password.value);
+    },
+    invalidMessage: "8文字以上の大小の英数字を交ぜたものにしてください。"
+  }
+};
+
+const isBlankInInput = (value) => value.trim() === "";
+
+const isLimitTextLength = (value, limit) => value.length >= limit;
+
+const isValidInRegex = (constraint, value) => constraint.test(value) ? false : true;
+
+
+const addValidClassName = (target) => {
+  const parent = target.parentElement;
+  parent.classList.add("valid");
+};
+
+const checkFieldValidation = (field) => {
```

Comment:
isValidField

Reply:
That way, I could tell from the function name that the return value was a Boolean value.

I changed the target function name.
[40b62d1](https://github.com/sae-github/JS_lesson/pull/30/commits/40b62d13951c04b8f6ad4f8f10f1a8e28d891767)